lib/models/pivtoken.js

```diff
@@ -0,0 +1,742 @@
+            operation: 'delete',
+            key: opts.params.guid
+        };
+
+        if (opts.etag) {
+            delPivOpts.options = {
+                etag: opts.etag
+            };
+        }
+
+        opts.moray.batch([delPivOpts, {
+            bucket: mod_recovery_token.bucket().name,
+            operation: 'deleteMany',
+            filter: util.format('(pivtoken=%s)', opts.params.guid)
+        }], function delBatchCb(delErr, delMeta) {
+            if (delErr) {
+                callback(delErr);
+                return;
+            }
+            opts.log.debug(delMeta, 'batch delete metadata');
+            callback(null);
+        });
+    }
+    validate.params(DELETE_SCHEMA, null, opts.params, deleteCb);
+}
+
+function listPIVTokens(opts, callback) {
+    assert.object(opts, 'opts');
+    assert.object(opts.moray, 'opts.moray');
+    assert.object(opts.log, 'opts.log');
+    assert.object(opts.params, 'opts.params');
+    assert.func(callback, 'callback');
+
+    opts.log.debug({ params: opts.params }, 'listTokens: entry');
+
+    model.list(Object.assign(opts, {
+        bucket: BUCKET,
+        validFields: VALID_FIELDS,
+        sort: {
+            attribute: 'guid',
+            order: 'ASC'
+        },
+        model: PIVToken,
+        defaultFilter: '(guid=*)',
+        listCb: function listObjsCb(mErr, pivtokens) {
+            if (mErr) {
+                callback(mErr);
+                return;
+            }
+
+            if (!pivtokens.length) {
+                callback(null, []);
+                return;
+            }
+
+            // Fetch all the recovery tokens associated with the PIVTokens
+            // retrieved by the previous call:
+            var pivGuids = pivtokens.map(function (p) {
+                return '(pivtoken=' + p.guid + ')';
+            }).join('');
+            var multiFilter = '(|' + pivGuids + ')';
+            mod_recovery_token.ls({
+                params: {
+                    filter: multiFilter
+                },
+                moray: opts.moray,
+                log: opts.log
+            }, function (lErr, tokens) {
+                if (lErr) {
+                    callback(lErr);
+                    return;
+                }
+
+                pivtokens = pivtokens.map(function addTokens(aPiv) {
+                    stripSensitiveFields(aPiv);
+                    var ownedTokens = tokens.filter(function filterTk(t) {
+                        return t.params.pivtoken === aPiv.params.guid;
+                    });
+
+                    aPiv.params.recovery_tokens = ownedTokens.map(
+                        function ser(t) {
+                        delete t.params.pivtoken;
+                        delete t.params.token;
+                        return t.serialize();
+                    });
+
+                    return aPiv;
+                });
+
+                callback(null, pivtokens);
+            });
+        }
+    }), callback);
```

Comment:
Actually, the only thing the listCb is doing is loading all the recovery tokens associated to every pivtokens fetched by the listing and striping sensitive fields. I can call `listCb` something else like `prepareResults` or whatever else if it might improve readability.

Reply:
No, it's more than just the naming - if you look at `ListModels` in lib/models/model.js, the callback behaviour is changing depending upon whether a listCb is passed in (this is the red flag)- sometimes cb() is called, othertimes listCb() is called - but not both.

I'd much prefer to see the loading of the recovery tokens to be another step, so maybe remove listCb, have lib/models/pivtoken.js call ListModels, then when it returns go and load the recovery tokens for each pivmodel returned.

Reply:
No more `listCb`. Just the same `cb` used here and there.